### PR TITLE
Add a flag to make get[Process|System]CpuLoad() match RI behaviour

### PR DESCRIPTION
**Pull request by thallium**
Currently, getProcessCpuLoad() and getSystemCpuLoad() return -1 if only one data point has been recorded. A compatibility flag -XX:[+/-]CpuLoadCompatibility is added to match the behaviour of the RI, which is to return 0.

Fixes: https://github.com/eclipse-openj9/openj9/issues/13389
Related: https://github.com/eclipse/omr/pull/7189

### CODE REVIEW

**Comment by thallium** [comment]
@TobiAjila FYI

**Comment by tajila** [comment]
@ChengJin01 Please review these changes

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java [hunk 1]
```diff
@@ -245,6 +245,12 @@ public final String getHardwareModel() throws UnsupportedOperationException {
 	/* Returns the number of CPU's online at this very moment. */
 	private native int getOnlineProcessorsImpl();
 
+	/**
+	 * Check if the CpuLoadCompatibility flag is set
```
Comment (ChengJin01) [inline jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java:249]: Please add the period `.` at the end of the description and the other comments in the PR.

**Review by ChengJin01 — APPROVED** [review]
Approve the PR from code perspective assuming the description & comments are appended with `.`.

File: runtime/jcl/common/mgmtosext.c [hunk 2]
```diff
@@ -96,7 +98,10 @@ Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSyste
 			portLibraryStatus = -3;
 			break;
 		default:
-			portLibraryStatus = OMRPORT_ERROR_OPFAILED;
+			portLibraryStatus =
```
Comment (tajila) [inline runtime/jcl/common/mgmtosext.c:101]: were you able to verify that this case is only possible if it is the first call?

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java (hunk 1 above)
Comment (thallium) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: Done

File: runtime/jcl/common/mgmtosext.c (hunk 2 above)
Comment (thallium) [reply runtime/jcl/common/mgmtosext.c:101]: Yes if the time tracking mechanism works correctly.

Comment (tajila) [reply runtime/jcl/common/mgmtosext.c:101]: In the case it doesnt work correctly we still need to return a negative number

Comment (thallium) [reply runtime/jcl/common/mgmtosext.c:101]: So `omrsysinfo_get_CPU_load()` need to return another error code?

Comment (tajila) [reply runtime/jcl/common/mgmtosext.c:101]: yes, if there is more than one way to return `OMRPORT_ERROR_OPFAILED`

Comment (thallium) [reply runtime/jcl/common/mgmtosext.c:101]: I've made a change to OMR to return another error code on first call https://github.com/eclipse/omr/pull/7189

**Comment by tajila** [comment]
@thallium Please rebase this PR

**Comment by thallium** [comment]
@TobiAjila Rebased

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk17

**Comment by tajila** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
@thallium there are some test failures

**Comment by thallium** [comment]
@tajila There's a related [test](https://github.com/eclipse-openj9/openj9/blob/master/test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOperatingSystemMXBean.java#L324) testing the return value of first call. Should we change it to zero or remove it?

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java [hunk 3]
```diff
@@ -263,16 +269,21 @@ public final synchronized double getProcessCpuLoad() {
...
+		 * return -1 for compatibility.
 		 */
 		if (-1 == oldTime) {
 			/* Save current counters; next invocation onwards, we use these to
 			 * compute CPU loads.
 			 */
 			oldTime = interimTime = latestTime;
 			oldCpuTime = interimCpuTime = latestCpuTime;
-			return CpuLoadCalculationConstants.ERROR_VALUE;
+			if (hasCpuLoadCompatibilityFlag()) {
```
Comment (tajila) [inline jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java:282]: this check is backwards

Comment (tajila) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java:282]: we should return zero if compatibility flag is set

File: runtime/jcl/common/mgmtosext.c [hunk 4]
```diff
@@ -95,6 +97,12 @@ Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSyste
 		case OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED:
 			portLibraryStatus = -3;
 			break;
+		case OMRPORT_ERROR_INSUFFICIENT_DATA:
+			portLibraryStatus =
+				J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_CPU_LOAD_COMPATIBILITY)
```
Comment (tajila) [inline runtime/jcl/common/mgmtosext.c:102]: same here

**Comment by tajila** [comment]
See https://github.com/eclipse-openj9/openj9/issues/13389#issuecomment-920219004

**Comment by thallium** [comment]
Fixed the logic and commit message.

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk17

**Comment by tajila** [comment]
@thallium Can you please open a docs issue

**Comment by pshipton** [comment]
Reverted via https://github.com/eclipse-openj9/openj9/pull/18635

**Pull request merged**